openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +388,226 @@ def debug_cohort_mgmt(request, course_key_string):
+
+
+class CohortHandler(DeveloperErrorViewMixin, APIPermissions):
+    """
+    **Use Cases**
+
+        Get the current cohorts in a course.
+        Create a new cohort in a course.
+        Modify a cohort in a course.
+
+    **Example Requests**:
+
+        GET /api/cohorts/v1/courses/{course_id}/cohorts
+        POST /api/cohorts/v1/courses/{course_id}/cohorts
+        GET /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}
+        PATCH /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}
+
+    **Response Values**
+
+        * cohorts: List of cohorts.
+        * cohort: A cohort representation:
+            * name: The string identifier for a cohort.
+            * id: The integer identifier for a cohort.
+            * user_count: The number of students in the cohort.
+            * assignment_type: The string representing the assignment type.
+            * user_partition_id: The integer identified of the UserPartition.
+            * group_id: The integer identified of the specific group in the partition.
+    """
+
+    def get(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to get either one or all cohorts.
+        """
+        course_key, course = api.get_course(request, course_key_string)
+        if not cohort_id:
+            all_cohorts = cohorts.get_course_cohorts(course)
+            paginator = NamespacedPageNumberPagination()
+            paginator.max_page_size = 100
+            page = paginator.paginate_queryset(all_cohorts, request)
+            response = [_get_cohort_representation(c, course) for c in page]
+            return Response(response, status=status.HTTP_200_OK)
+        else:
+            cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
+            return Response(_get_cohort_representation(cohort, course), status=status.HTTP_200_OK)
+
+    def post(self, request, course_key_string, cohort_id=None):
+        """
+        Endpoint to create a new cohort, must not include cohort_id.
+        """
+        if cohort_id is not None:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Please user the parent endpoint.',
+                                 'wrong-endpoint')
+        course_key, course = api.get_course(request, course_key_string)
+        name = request.data.get('name')
+        if not name:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Cohort name must be specified.',
+                                 'missing-cohort-name')
+        assignment_type = request.data.get('assignment_type')
+        if not assignment_type:
+            raise self.api_error(status.HTTP_400_BAD_REQUEST,
+                                 'Assignment type must be specified.',
+                                 'missing-assignment-type')
+        return Response(_get_cohort_representation(cohorts.add_cohort(course_key, name, assignment_type), course))
```

Comment:
This is a bit hard to follow. Could you break it into two lines?

Since this code is common it could be put into a method called `_get_cohort_response` or something.

Reply:
👍